### DISCUSSION TITLE
Blog post "how-to-implement-a-basic-activitypub-server" misses the field `Digest`

### DISCUSSION THREAD

**Discussion by green-coder** [mastodon/mastodon#28009] [General]
### Steps to reproduce the problem

1. Read the blog article at https://blog.joinmastodon.org/2018/06/how-to-implement-a-basic-activitypub-server/
2. Try to send a POST request to a Mastodon instance.

### Expected behaviour

The POST request should be accepted by the server (code 202)

### Actual behaviour

The server answers an error 401

### Specifications

The blog post is missing the information about the `Digest` field.

See https://github.com/mastodon/mastodon/blob/c3aef491d66aec743a3a53e934a494f653745b61/app/controllers/concerns/signature_verification.rb#L131

**Comment by mfosterio** [comment]
How should the Digest field look?

> ### Steps to reproduce the problem
> 1. Read the blog article at https://blog.joinmastodon.org/2018/06/how-to-implement-a-basic-activitypub-server/
> 2. Try to send a POST request to a Mastodon instance.
> 
> ### Expected behaviour
> The POST request should be accepted by the server (code 202)
> 
> ### Actual behaviour
> The server answers an error 401
> 
> ### Specifications
> The blog post is missing the information about the `Digest` field.
> 
> See
> 
> https://github.com/mastodon/mastodon/blob/c3aef491d66aec743a3a53e934a494f653745b61/app/controllers/concerns/signature_verification.rb#L131

**Comment by green-coder** [comment]
> How should the Digest field look?

It's written in those lines: https://github.com/mastodon/mastodon/blob/c3aef491d66aec743a3a53e934a494f653745b61/app/controllers/concerns/signature_verification.rb#L134-L142

The digest field should be added to the "signed string" like that (in Clojure):  https://codeberg.org/green-coder/the-pub/src/branch/made-in-taiwan/src/the_pub/ring_handler.clj#L95-L98

The field `digest` should be added in the signature field like that (in Clojure): https://codeberg.org/green-coder/the-pub/src/branch/made-in-taiwan/src/the_pub/ring_handler.clj#L103

This commit diff sums up the information missing in the blog, which I found in the Mastodon source code:
https://codeberg.org/green-coder/the-pub/commit/1da6f95efa1918efa9e9eedba929df46f591b559

**Comment by sventiffe** [comment]
Can/will old blog posts be updated? 
See PR for security.md: https://github.com/mastodon/documentation/pull/1082
And issue filed: https://github.com/mastodon/documentation/issues/1079 (I did not check issues in mastodon/mastodon before filing it as it's a separate repo and I didn't investigate where blog posts are defined)
